pkg/proxy/iptables/proxier.go

```diff
@@ -308,6 +313,16 @@ func NewProxier(ipt utiliptables.Interface,
 		klog.Warning("missing br-netfilter module or unset sysctl br-nf-call-iptables; proxy may not work as intended")
 	}
 
+	masqueradeRandomFully := false
+	iptVersionString, err := ipt.GetVersion()
+	if err == nil {
+		iptVersion, err := utilversion.ParseGeneric(iptVersionString)
```

Comment:
I am having trouble understanding the words here; this line invokes a different function than line 317.  Can you please elaborate on the problem?

Reply:
I got my lines confused. I meant to comment on L318-328 where we run ParseGeneric a few times in different nested ifs. Can we consolidate these lines and utilize something like [version.Compare](https://github.com/kubernetes/apimachinery/blob/master/pkg/util/version/version.go#L308). It might just be me but I got a little lost walking through all of these nested conditions. I also had to reread the final comparison on L326 a few times before I understood that we are setting that flag to true based on the other conditions. 

Perhaps @dcbw or @danwinship will have other thoughts on if it looks ok to them.

Reply:
@cmluciano : in lines 318--328 there are two different strings parsed with `utilversion.ParseGeneric`.  We coud remove one of those by using `version.Compare` --- which does a parse internally --- instead of `version.AtLeast`.

Reply:
@danwinship : it looks like https://github.com/kubernetes/kubernetes/pull/78547#discussion_r296884083 was intended to be attached here, right?

I am a little surprised that `pkg/util/iptables` does not cache the version string.  Is that deliberate preparation for coping with live kernel patching?  Should we continue to eschew caching the version string and things derived from it for that reason?

There is no direct external consumer for the "supports check" bit, so "SupportsRandomFully" (data bit or query function) would be the first of its kind.

Reply:
> @danwinship : it looks like [#78547 (comment)](https://github.com/kubernetes/kubernetes/pull/78547#discussion_r296884083) was intended to be attached here, right?

oops. yes

> I am a little surprised that `pkg/util/iptables` does not cache the version string. Is that deliberate preparation for coping with live kernel patching?

Nothing so planned. Originally it was only needed in one place. Later whoever added the second usage must have felt it would be easier to just call the "get version" function again rather than cache it. it's not like it takes that long to run "`iptables --version`", and you only end up doing it a few times anyway...

> There is no direct external consumer for the "supports check" bit, so "SupportsRandomFully" (data bit or query function) would be the first of its kind.

There are external consumers (like `pkg/proxy/iptables/proxier.go`), they just have to implement it by getting the iptables version and manually comparing it against `utiliptables.MinCheckVersion`...

Reply:
I have refactored, consolidating the version string parsing and comparison logic.